src/backend/cdb/cdbtm.c

```diff
@@ -1745,12 +1745,8 @@ setupQEDtxContext(DtxContextInfo *dtxContextInfo)
 				LWLockAcquire(SharedLocalSnapshotSlot->slotLock, LW_SHARED);
 				elog(DTM_DEBUG5,
 					 "setupQEDtxContext inputs (part 2b):  shared local snapshot xid = " UINT64_FORMAT " "
-					 "(xmin: %u xmax: %u xcnt: %u) curcid: %d, QDxid = "UINT64_FORMAT"/%u",
+					 ", QDxid = "UINT64_FORMAT"/%u",
 					 U64FromFullTransactionId(SharedLocalSnapshotSlot->fullXid),
-					 SharedLocalSnapshotSlot->snapshot.xmin,
```

Comment:
If we don't need SharedLocalSnapshotSlot->snapshot, remove it?

Reply:
Yes, the snapshot has been removed from SharedLocalSnapshotSlot, but the slot still needs to be kept. As you can see, several status variables and the process related context here
```
typedef struct SharedSnapshotSlot
{
	int32			slotindex;  /* where in the array this one is. */
	int32	 		slotid;
	PGPROC			*writer_proc;

	/* only used by cursor dump identification, dose not always set */
	volatile DistributedTransactionId distributedXid;

	volatile bool			ready;
	volatile uint32			segmateSync;

	volatile dsm_handle		snapshot_handle;
	uint32					snapshot_sz;
	LWLock					*slotLock;

	volatile int			cur_dump_id;
	volatile SnapshotDump	dump[SNAPSHOTDUMPARRAYSZ];
	/* for debugging only */
	FullTransactionId		fullXid;
	TimestampTz				startTimestamp;
} SharedSnapshotSlot;
```